puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/foreman-proxy-develop-test.groovy

```diff
@@ -0,0 +1,89 @@
+def commit_hash = ''
+
+pipeline {
+    agent any
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage("Collect Git Hash") {
+            steps {
+                git(url: 'https://github.com/theforeman/smart-proxy', branch: 'develop')
+                script {
+                    commit_hash = archive_git_hash()
+                }
+            }
+        }
+        stage("test ruby-2.1 & puppet-4.10.9") {
+            steps {
+                run_test(ruby: '2.1', puppet: '4.10.9')
+            }
+        }
+        stage("test ruby-2.2 & puppet-4.10.9") {
+            steps {
+                run_test(ruby: '2.2', puppet: '4.10.9')
+            }
+        }
+        stage("test ruby-2.3 & puppet-4.10.9") {
```

Comment:
Testing with Puppet 3.8.7 would actually make more sense here since that's what Xenial runs out of the box. With Puppet 4 we don't use the actual gem anyway.